src/runtime/string.cpp

```diff
@@ -646,6 +646,30 @@ static PRIMFN(prim_str2bin) {
   RETURN(Integer::alloc(runtime.heap, out));
 }
 
+static PRIMTYPE(type_str2bytes) {
+  TypeVar list;
+  Data::typeList.clone(list);
+  list[0].unify(Data::typeInteger);
+  return args.size() == 1 && args[0]->unify(Data::typeString) && out->unify(list);
+}
+
+static PRIMFN(prim_str2bytes) {
+  EXPECT(1);
+  STRING(arg0, 0);
+
+  // worst-case estimate
+  size_t need = reserve_list(arg0->size()) + arg0->size() * Integer::reserve(1);
```

Comment:
Mind explaining this a bit more?  (Maybe even in the comment?)  Since we're on the C++ side and from looking at what's below I assume we're already safely dealing with the UTF8 representation rather than Unicode code points, but I don't know exactly what all the `reserve`s correspond to.  Calculate the size of the overhead plus the maximum size of an Integer, both for every list item?

Reply:
I'm mostly going off other code reference to derive this value (actually one of the reasons I put @terpstra on the review) but my vague understanding is the following

`reserve_list(arg0->size())` -> Reserve enough memory for a list of `size` elements
`arg0->size() * Integer::reserve(1)` -> Reserve enough memory for `size` number of `Integer`s. Might actually be sufficient to instead do `Integer::reserve(arg0->size())` but I'm not confident enough to assert that.